tdp/core/models/stale_component.py

```diff
@@ -77,28 +85,30 @@ def generate(
         )
         stale_components_dict = {}
         for operation in config_and_restart_operations:
-            key = (operation.service_name, operation.component_name)
-            stale_component = stale_components_dict.setdefault(
-                key,
-                StaleComponent(
-                    service_name=operation.service_name,
-                    component_name=operation.component_name
-                    if not operation.is_service_operation()
-                    else "",
-                    to_reconfigure=False,
-                    to_restart=False,
-                ),
-            )
-            if operation.action_name == "config":
-                stale_component.to_reconfigure = True
-            if operation.action_name == "restart":
-                stale_component.to_restart = True
+            for host_name in operation.host_names:
```

Comment:
`operation.host_names` can be `None` no ? For `noop` operation this is `None`.

Reply:
Yes, `operation.hot_names` can be an empty set. I fixed it!